packages/notification-center/src/components/notification-center/components/user-preference/ChannelPreference.tsx

```diff
@@ -1,17 +1,45 @@
 import React from 'react';
-import { Text, TextBlock } from './UserPreference';
 import { useNovuThemeProvider } from '../../../../hooks/use-novu-theme-provider.hook';
 import { Switch } from '@mantine/core';
 import { getChannel } from './channels';
 import styled from 'styled-components';
+import { useApi } from '../../../../hooks';
+import { switchStyles, Text, TextBlock } from './styles';
+import { IUserPreferenceSettings } from '../../../../index';
 
-export function ChannelPreference({ type, active }) {
+interface IChannelPreferenceProps {
+  type: string;
+  active: boolean;
+  templateId: string;
+  index: number;
+  setSettings: (IUserPreferenceSettings) => void;
+}
+export function ChannelPreference({ type, active, templateId, index, setSettings }: IChannelPreferenceProps) {
   const { label, description, Icon } = getChannel(type);
   const { theme } = useNovuThemeProvider();
+  const { api } = useApi();
   const baseTheme = theme?.notificationItem?.unseen;
   const primaryColor = baseTheme.fontColor;
   const secondaryColor = baseTheme.timeMarkFontColor;
 
+  const handleUpdateChannelPreference = async (checked) => {
+    const result = await api.updateSubscriberPreference(templateId, type, checked);
+
+    setSettings((prev) =>
+      prev.map((workflow, i) => {
+        return i === index
+          ? {
+              template: workflow.template,
+              preference: {
+                ...workflow.preference,
+                channels: { ...workflow.preference.channels, [type]: result.channels[type] },
+              },
+            }
+          : workflow;
```

Comment:
Maybe worth converting this to just a regular if/statement it might be easier to read here.